auth/src/test/java/org/aerogear/auth/credentials/OIDCCredentialsTest.java

```diff
@@ -0,0 +1,125 @@
+        assertFalse("Expect Token Validation with Wrong Audience to Fail", oidcCredentials.verifyToken(VALID_ACCESS_TOKEN, VALID_PUBLIC_KEY, VALID_ISSUER, INVALID_AUDIENCE));
+    }
+
+    @Test
+    public void testValidIssuer() {
+        OIDCCredentials oidcCredentials = new OIDCCredentials();
+        assertTrue("Expect Token Validation with Valid Issuer to Succeed", oidcCredentials.verifyToken(VALID_ACCESS_TOKEN, VALID_PUBLIC_KEY, VALID_ISSUER, VALID_AUDIENCE));
+    }
+
+    @Test
+    public void testInvalidIssuer() {
+        OIDCCredentials oidcCredentials = new OIDCCredentials();
+        assertFalse("Expect Token Validation with Wrong Issuer to Fail", oidcCredentials.verifyToken(VALID_ACCESS_TOKEN, VALID_PUBLIC_KEY, INVALID_ISSUER, VALID_AUDIENCE));
+    }
+
+    @Test
+    public void testExpiredExpirationTime() {
+        OIDCCredentials oidcCredentials = new OIDCCredentials();
+        assertFalse("Expect Token Validation with Expired Time to Fail", oidcCredentials.verifyToken(EXPIRED_ACCESS_TOKEN, VALID_PUBLIC_KEY, VALID_ISSUER, VALID_AUDIENCE));
+    }
+
+    @Test
+    public void testValidAccessToken() {
+        OIDCCredentials oidcCredentials = new OIDCCredentials();
+        assertTrue("Expect Token Validation with Valid Access Token to Succeed", oidcCredentials.verifyToken(VALID_ACCESS_TOKEN, VALID_PUBLIC_KEY, VALID_ISSUER, VALID_AUDIENCE));
+    }
+
+    @Test
+    public void testTamperedAccessToken() {
+        OIDCCredentials oidcCredentials = new OIDCCredentials();
+        assertFalse("Expect Token Validation with Tampered Access Token to Fail", oidcCredentials.verifyToken(TAMPERED_ACCESS_TOKEN, VALID_PUBLIC_KEY, VALID_ISSUER, VALID_AUDIENCE));
+    }
+
+    @Test
+    public void testValidIdentityToken() {
+        OIDCCredentials oidcCredentials = new OIDCCredentials();
+        assertTrue("Expect Token Validation with Valid Identity Token to Succeed", oidcCredentials.verifyToken(VALID_IDENTITY_TOKEN, VALID_PUBLIC_KEY, VALID_ISSUER, VALID_AUDIENCE));
+    }
+
+    @Test
+    public void testTamperedIdentityToken() {
+        OIDCCredentials oidcCredentials = new OIDCCredentials();
+        assertFalse("Expect Token Validation with Tampered Identity Token to Fail", oidcCredentials.verifyToken(TAMPERED_IDENTITY_TOKEN, VALID_PUBLIC_KEY, VALID_ISSUER, VALID_AUDIENCE));
+    }
+
+    @Test
+    public void testValidRefreshToken() {
+        OIDCCredentials oidcCredentials = new OIDCCredentials();
+        assertTrue("Expect Token Validation with Valid Refresh Token to Succeed", oidcCredentials.verifyToken(VALID_REFRESH_TOKEN, VALID_PUBLIC_KEY, VALID_ISSUER, VALID_AUDIENCE));
+    }
+
+    @Test
+    public void testTamperedRefreshToken() {
+        OIDCCredentials oidcCredentials = new OIDCCredentials();
+        assertFalse("Expect Token Validation with Tampered Refresh Token to Fail", oidcCredentials.verifyToken(TAMPERED_REFRESH_TOKEN, VALID_PUBLIC_KEY, VALID_ISSUER, VALID_AUDIENCE));
+    }
+
```

Comment:
Could the repeated "OIDCCredentials oidcCredentials = new OIDCCredentials();" be in the setup block?